### PR TITLE
reduxless module and  cloud to device message

### PR DESCRIPTION
**Pull request by asrudra**

### CODE REVIEW

File: src/app/devices/deviceContent/cloudToDeviceMessage/saga.ts [hunk 1]
```diff
@@ -65,3 +63,7 @@ export function* cloudToDeviceMessageSaga(action: Action<CloudToDeviceMessageAct
         yield put(cloudToDeviceMessageAction.failed({params: action.payload, error}));
     }
 }
+
+export function* cloudToDeviceMessageSagaWatcher() {
```
Comment (YingXue) [inline src/app/devices/deviceContent/cloudToDeviceMessage/saga.ts:67]: I am curious about the rule that we shall follow in all sagas, worker/watch/normal saga. Compared with saga.ts in addDevice

File: src/app/devices/deviceContent/cloudToDeviceMessage/saga.ts [hunk 2]
```diff
@@ -2,9 +2,9 @@
  * Copyright (c) Microsoft Corporation. All rights reserved.
  * Licensed under the MIT License
  **********************************************************/
-import { call, put } from 'redux-saga/effects';
+import { call, put, takeEvery } from 'redux-saga/effects';
 import { Action } from 'typescript-fsa';
-import { cloudToDeviceMessageAction, CloudToDeviceMessageActionParameters } from '../actions';
+import { cloudToDeviceMessageAction, CloudToDeviceMessageActionParameters } from './actions';
 import { getActiveAzureResourceConnectionStringSaga } from '../../../azureResource/sagas/getActiveAzureResourceConnectionStringSaga';
```
Comment (YingXue) [inline src/app/devices/deviceContent/cloudToDeviceMessage/saga.ts:8]: an unused import

File: src/app/devices/deviceContent/cloudToDeviceMessage/saga.ts (hunk 1 above)
Comment (asrudra) [reply src/app/devices/deviceContent/cloudToDeviceMessage/saga.ts:67]: agreed. Please set a rule. will follow

File: vim.exe.stackdump [hunk 3]
```diff
@@ -0,0 +1,19 @@
+Stack trace:
```
Comment (YingXue) [inline vim.exe.stackdump:1]: this file should not be included seems like

Comment (asrudra) [reply vim.exe.stackdump]: removed

File: src/app/devices/module/moduleIndentityDetail/state.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+ * Licensed under the MIT License
+ **********************************************************/
+import { Record } from 'immutable';
+import { IM } from '../../../shared/types/types';
+import { SynchronizationWrapper } from '../../../api/models/synchronizationWrapper';
+import { ModuleIdentity } from '../../../api/models/moduleIdentity';
+import { SynchronizationStatus } from '../../../api/models/synchronizationStatus';
+
+export interface ModuleIdentityDetailStateInterface extends SynchronizationWrapper<ModuleIdentity>{}
+
```
Comment (YingXue) [inline src/app/devices/module/moduleIndentityDetail/state.ts:12]: I like the way you have your state interface defined. I went with composition not inheritence that created an uncessary level. Will snap to yours once your change gets in.

File: src/app/devices/module/moduleIndentityDetail/sagas/deleteModuleIdentitySaga.spec.ts [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+import { call, put } from 'redux-saga/effects';
+import { deleteModuleIdentitySaga } from './deleteModuleIdentitySaga';
+import { deleteModuleIdentityAction } from '../actions';
+import * as ModuleService from '../../../../api/services/moduleService';
+import { ResourceKeys } from '../../../../../localization/resourceKeys';
+import { NotificationType } from '../../../../api/models/notification';
+import { ModuleIdentity } from '../../../../api/models/moduleIdentity';
+import { raiseNotificationToast } from '../../../../notifications/components/notificationToast';
+
+describe('moduleIdentitySaga', () => {
```
Comment (YingXue) [inline src/app/devices/module/moduleIndentityDetail/sagas/deleteModuleIdentitySaga.spec.ts:17]: would be deleteModuleIdentitySaga (might be copied from the old file which was wrong...)

**Pull request merged**